src/redux/books/booksSlice.js

```diff
@@ -1,42 +1,72 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const API = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/XKLrcF8yEfNG679PvMbv/books';
 
 const initialState = {
-  books: // Initial state:
-  [
-    {
-      item_id: 0,
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 1,
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 2,
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: [],
+  status: 'notStarted',
+  error: null,
 };
 
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  try {
+    const response = await axios(API);
+    const booksWithId = Object.entries(response.data).map(([id, book]) => ({ id, book: book[0] }));
+    return booksWithId;
+  } catch (error) {
+    throw new Error(error.message);
+  }
+});
+
+export const addBook = createAsyncThunk('books/addBook', async (newBook, thunkAPI) => {
+  try {
+    await axios.post(API, newBook);
+    const response = await thunkAPI.dispatch(fetchBooks());
+    return response;
+  } catch (error) {
+    throw new Error(error.message);
+  }
+});
+
+export const removeBook = createAsyncThunk('books/deleteBook', async (bookId) => {
+  try {
+    const response = await axios.delete(`${API}/${bookId}`);
+    return response.data;
+  } catch (error) {
+    throw new Error(error.message);
+  }
+});
```

Comment:
- [ ] Please fix your remove functionality because your remove functionality is not working when I try to remove any book it's removed but when it refreshed automatically then my book is still there so kindly fix :slightly_smiling_face: 

[OPTIONAL] Please fix your styling because I am unable to see the `comment`, `edit` and `remove` buttons so kindly fix it as well :+1:   

![Screenshot from 2023-05-28 03-18-52](https://github.com/emarkees/book-store/assets/45226848/599aac21-abf7-4ba3-bafb-7289e9e89847)

Reply:
@aamirkhan2478 can i have a session with you